### ISSUE TITLE
The Collection-Compound Class

### ISSUE THREAD

**Issue by OrtnerMichael** [magpylib/magpylib#444]
from #434 

The idea is to enable (but not add) the Compound class which refers to all classes that inherit BaseGeo AND Collection. This enables "tracking" of Collection objects that could be Halbach Arrays or similar magnet compounds.

For this to work the move/rotate operations have to be rewritten so that they do not overwrite each other but can apply to BaseGeo and Collection objects at the same time.

When `display(path='True')` only the compound path should be visible by default - how to turn on all paths ?

- [x] #443
- [x] fix display methods - how to turn all paths on and off ?
- [x] #446 
- [x] fix tests
- [x] do extended manual testing (@OrtnerMichael)
- [x] update docstrings (edit `Collection` docstring with Compound properties)
- [x] update changelog
- [x] update docs
- [x] #450
- [x] choose N!=M behavior

**Comment by Alexboiboi** [comment]
just some example code to show how it could look like

```python
import magpylib as magpy
import numpy as np
from magpylib import Collection
from magpylib._src.obj_classes.class_BaseGeo import BaseGeo


class CompoundClass(BaseGeo, Collection):
    def __init__(self, *args, position=(0.,0.,0.), orientation=None, style=None):
        BaseGeo.__init__(self, position=position, orientation=orientation, style=style)
        Collection.__init__(self, *args)


N1 = 8
N2 = 20
cs_lambda = lambda: magpy.magnet.Cuboid(
    (1, 0, 0), dimension=(10, 10, 10), position=(18, 0, 0)
)
s0 = cs_lambda().rotate_from_angax(
    [360 / (N1)] * N1, "z", increment=True, anchor=(0, 0, 0)
)

cs_list = []
for ind in range(N):
    s = cs_lambda()
    s.position = s0.position[ind]
    s.orientation = s0.orientation[ind]
    cs_list.append(s)
c = CompoundClass(cs_list).set_styles(magnetization_show=False, path_show=False)
c.rotate_from_angax(
    np.linspace(0, 720, N2), axis="z", start=-1, anchor=(80, 0, 0)
)
c.move(np.linspace((0,0,0), (0,0,200), N2), start=0)
c.style.color = 'red'
c.style.name = "Magnetic Wheel"
magpy.display(c, path=2, backend="plotly")
```

![image](https://user-images.githubusercontent.com/29252289/147890048-4b0608e2-c289-404c-9b8a-d318581375c0.png)

**Comment by Alexboiboi** [comment]
>  fix move and rotate

Could you please expound on what is to be fixed here?


> fix display methods - how to turn all paths on and off ?

partially answered in #446
> Compound class docstring example

How would you have a docstring and in the same time not add it to the library ?

**Comment by OrtnerMichael** [comment]
des ist geil 👯‍♂️ 

- I'm working on move and rotate fix according to our discussion #438 
- The unpacking with * makes a lot of sense !!! Cool way of implementing this !!!
- I didnt mean docstring - I meant doku example

**Comment by Alexboiboi** [comment]
> * I'm working on move and rotate fix according to our discussion

in which branch are you working?

**Comment by OrtnerMichael** [comment]
@Alexboiboi please comment on this:

There are two possibilities to interpret "ambiguous" input when
1. BaseGeo init with len(position_path)=M and len(orientation_path)=N and M!=N
2. Multi-anchor rotation when len(rotation_input)=M and len(anchor_input)=M and M!=N

Both cases should behave similarly in my opinion.

I propose that we follow our new "padding logic" (like when start>path_length):
"If one input is longer than the other, pad the other with its edge value"
 - Advantage: simple rule, simple code
 - Disadvantage: no warnings/error msg when this nonsensical input comes

Example:
```
src = magpy.Sensor(position=(1,2,3), orientation=R.from_rotvec((0,0,1),(0,0,2)
print(src.position)                         # [[1,2,3]  [1,2,3]]
print(src.orientation.as_rotvec())  # [[0,0,1]   [0,0,2]]
```

An alternative would be to raise a BadUserInput Exception when N!=M (unless the smaller one is 1).
 - Advantage: error msg
 - Disadvantage: more restrictive, more complex code
```
src = magpy.Sensor(position=(1,2,3), orientation=R.from_rotvec((0,0,1),(0,0,2)
# Error: MagpylibBadUserInput....
```
what so you think ?

**Comment by OrtnerMichael** [comment]
my move_rotate_compound tests
```
import numpy as np
import magpylib as magpy
from scipy.spatial.transform import Rotation as R

# position and orientation paths are tiled up at init

# An operation applied to a collection is individually
# applied to BaseGeo and to each child.

# BaseGeo is mostly used for tracking the Collection

# pylint: disable=no-member

def test_po(obj, ppath, opath_as_rotvec, pshape, oshape=None):
    """ test object position and orientation"""
    oshape = pshape if oshape is None else oshape
    sp = obj.position
    so = obj.orientation.as_rotvec()
    assert np.allclose(sp, ppath) and sp.shape==pshape
    assert np.allclose(so, opath_as_rotvec) and so.shape==oshape


###############################################################################
# INIT OBJECT TESTING

# at initialization position and orientation to have similar shape (N,3)
# - if inputs are (N,3) and (3,) then the (3,) is tiled up to (N,3)
# - if inputs are (N,3) and (M,3) then the smaller one is padded up
# - None orientation input is interpreted as (0,0,0) rotvec / (0,0,0,1) quat

p0 = (1,2,3)
p1 = [(1,2,3)]
p2 = [(1,2,3), (1,1,1)]
o0 = None
o1 = (0,0,.1)
o2 = [(0,0,.1)]
o3 = [(0,0,.1), (0,0,.2)]
o4 = [(0,0,.1), (0,0,.2), (0,0,.3)]

# p, o, shape, ptest, otest
tests = [
    [p0, o0, (3,), p0, (0,0,0)],
    [p0, o1, (3,), p0, o1],
    [p0, o2, (3,), p0, o1],
    [p0, o3, (2,3), (p0,p0), o3],
    [p1, o0, (3,), p0, (0,0,0)],
    [p1, o1, (3,), p0, o1],
    [p1, o2, (3,), p0, o1],
    [p1, o3, (2,3), (p0,p0), o3],
    [p2, o0, (2,3), p2, [(0,0,0)]*2],
    [p2, o1, (2,3), p2, [o1]*2],
    [p2, o2, (2,3), p2, [o1]*2],
    [p2, o3, (2,3), p2, o3],
    [p2, o4, (3,3), p2+[(1,1,1)], o4]  # uneven paths
    ]

for test in tests:
    p, o, shape, ptest, otest = test

    o = (0,0,0) if o is None else o
    r = R.from_rotvec(o)
    src = magpy.magnet.Cuboid((1,0,0), (1,1,1), position=p, orientation=r)
    test_po(src, ptest, otest, shape)


############################################################################
# SETTER OBJECT TESTING

# object position setting
# - when position or orientation are set by user, the other attribute is not touched
#   advantage: the user is not aware of this happening and possibly wants to access
#              what he thinks is in the orientation path. Any auto-tiling requires
#              a complicated set of rules
# -> possible extended auto-tiling functionality - possibly v4.1 ?:
#    - if one is set to shape (N,3) and the other is (3,) then the other is tiled up
#    - what happens if one is set to (N,3) the other is still (M,3) for M<N and M>N ?

# when setting a position the orientation remains unchanged
src = magpy.magnet.Cuboid((1,0,0), (1,1,1), [(1,1,1),(2,2,2)])
test_po(src, [(1,1,1),(2,2,2)], [(0,0,0), (0,0,0)], (2,3))
src.position = (0,0,1)
test_po(src, (0,0,1), [(0,0,0), (0,0,0)], (3,), (2,3))

# when setting an orientation the position remains unchanged
src = magpy.magnet.Cuboid((1,0,0), (1,1,1), [(1,1,1),(2,2,2)])
test_po(src, [(1,1,1),(2,2,2)], [(0,0,0), (0,0,0)], (2,3))
src.orientation = R.from_rotvec((0,0,.1))
test_po(src, [(1,1,1),(2,2,2)], (0,0,.1), (2,3), (3,))


###############################################################################
# OBJECT MOVE AND ROTATION TESTS
# these are thoroly tested in previous commits
src = magpy.magnet.Cuboid((1,0,0), (1,1,1))
src.move([(1,1,1)])
test_po(src, [(0,0,0),(1,1,1)], [(0,0,0), (0,0,0)], (2,3))


src.rotate_from_rotvec((0,0,.1))
test_po(src, [(0,0,0),(1,1,1)], [(0,0,.1), (0,0,.1)], (2,3))


###############################################################################
# OBJECT MULTI-ANCHOR TESTING
# scalar path - scalar anchor
src = magpy.magnet.Cuboid((1,0,0), (1,1,1))
src.rotate_from_angax(90 , 'z', 0)
test_po(src, (0,0,0), (0,0,np.pi/2), (3,))

# vector path 1 - scalar anchor
src = magpy.magnet.Cuboid((1,0,0), (1,1,1))
src.rotate_from_angax([90] , 'z', (1,0,0))
test_po(src, [(0,0,0), (1,-1,0)], [(0,0,0),(0,0,np.pi/2)], (2,3,))

# vector path 2 - scalar anchor
src = magpy.magnet.Cuboid((1,0,0), (1,1,1))
src.rotate_from_angax([90, 270] , 'z', (1,0,0))
test_po(src, [(0,0,0), (1,-1,0), (1,1,0)], [(0,0,0),(0,0,np.pi/2),(0,0,-np.pi/2)], (3,3,))

# scalar path - vector anchor 1
src = magpy.magnet.Cuboid((1,0,0), (1,1,1))
src.rotate_from_angax(90 , 'z', [(1,0,0)])
test_po(src, [(0,0,0), (1,-1,0)], [(0,0,0),(0,0,np.pi/2)], (2,3,))

# scalar path - vector anchor 2
src = magpy.magnet.Cuboid((1,0,0), (1,1,1))
src.rotate_from_angax(90 , 'z', [(1,0,0),(2,0,0)])
test_po(src, [(0,0,0), (1,-1,0), (2,-2,0)], [(0,0,0),(0,0,np.pi/2),(0,0,np.pi/2)], (3,3,))

# vector path 2 - vector anchor 2
src = magpy.magnet.Cuboid((1,0,0), (1,1,1))
src.rotate_from_angax([90,270] , 'z', [(1,0,0),(2,0,0)])
test_po(src, [(0,0,0), (1,-1,0), (2,2,0)], [(0,0,0),(0,0,np.pi/2),(0,0,-np.pi/2)], (3,3,))

# vector path 2 - vector anchor 2 - path 2 - start=0
src = magpy.magnet.Cuboid((1,0,0), (1,1,1), position=[(0,0,0),(2,1,0)])
src.rotate_from_angax([90,270] , 'z', [(1,0,0),(2,0,0)], start=0)
test_po(src, [(1,-1,0), (3,0,0)], [(0,0,np.pi/2),(0,0,-np.pi/2)], (2,3,))

# init crazy, path 2, anchor 3, start middle
src = magpy.magnet.Cuboid((1,0,0), (1,1,1), position=[(0,0,0),(2,0,0)], orientation=R.from_rotvec((0,0,.1)))
src.rotate_from_angax([90,270] , 'z', anchor=[(1,0,0),(2,0,0),(3,0,0)], start=1)
test_po(
    src,
    [(0,0,0), (1,1,0), (2,0,0), (3,1,0)],
    [(0,0,.1), (0,0,np.pi/2+.1), (0,0,-np.pi/2+.1), (0,0,-np.pi/2+.1)],
    (4,3,))

# init crazy, path 2, anchor 3, start before
src = magpy.magnet.Cuboid((1,0,0), (1,1,1), position=[(0,0,0),(2,0,0)], orientation=R.from_rotvec((0,0,.1)))
src.rotate_from_angax([90,270] , 'z', anchor=[(1,0,0),(2,0,0),(3,0,0)], start=-4)
test_po(
    src,
    [(1,-1,0), (2,2,0), (3,3,0), (2,0,0)],
    [(0,0,.1+np.pi/2), (0,0,.1-np.pi/2), (0,0,.1-np.pi/2), (0,0,.1)],
    (4,3,))



##############################################################################
# COLLECTION - COMPOUND BEHAVIOR TESTS

# init Collection should not change source pos and ori
src = magpy.magnet.Cuboid((1,0,0), (1,1,1), [(1,2,3),(2,3,4)])
test_po(src, [(1,2,3),(2,3,4)], [(0,0,0)]*2, (2,3))
col = magpy.Collection(src, position=[(1,1,1)])
test_po(src, [(1,2,3),(2,3,4)], [(0,0,0)]*2, (2,3))

# test: An operation move() or rotate() applied to a Collection is
# individually applied to BaseGeo and to each child:

# very sensible Compound behavior with rotation anchor
s1 = magpy.magnet.Cuboid((1,0,0), (1,1,1), (1,1,1))
s2 = magpy.magnet.Cuboid((1,0,0), (1,1,1), (-1,-1,-1))
col = magpy.Collection(s1, s2)
col.move((0,0,1))
test_po(s1, (1,1,2), (0,0,0), (3,))
test_po(s2, (-1,-1,0), (0,0,0), (3,))
test_po(col, (0,0,1), (0,0,0), (3,))
col.move([(0,0,1)])
test_po(s1, [(1,1,2),(1,1,3)], [(0,0,0)]*2, (2,3))
test_po(s2, [(-1,-1,0),(-1,-1,1)], [(0,0,0)]*2, (2,3))
test_po(col, [(0,0,1),(0,0,2)], [(0,0,0)]*2, (2,3))
col.rotate_from_rotvec((0,0,np.pi/2), anchor=0)
test_po(s1, [(-1,1,2),(-1,1,3)], [(0,0,np.pi/2)]*2, (2,3))
test_po(s2, [(1,-1,0),(1,-1,1)], [(0,0,np.pi/2)]*2, (2,3))
test_po(col, [(0,0,1),(0,0,2)], [(0,0,np.pi/2)]*2, (2,3))

# very sensible Compound behavior with vector anchor
s1 = magpy.magnet.Cuboid((1,0,0), (1,1,1), (1,0,1))
s2 = magpy.magnet.Cuboid((1,0,0), (1,1,1), (-1,0,-1))
col = magpy.Collection(s1, s2, position=(3,0,3))
col.rotate_from_rotvec((0,0,np.pi/2), anchor=[(1,0,0), (2,0,0)])
test_po(s1, [(1,0,1),(1,0,1),(2,-1,1)], [(0,0,0), (0,0,np.pi/2), (0,0,np.pi/2)], (3,3))
test_po(s2, [(-1,0,-1),(1,-2,-1),(2,-3,-1)], [(0,0,0),(0,0,np.pi/2),(0,0,np.pi/2)], (3,3))
test_po(col, [(3,0,3), (1,2,3),(2,1,3)], [(0,0,0), (0,0,np.pi/2), (0,0,np.pi/2)], (3,3))

# very sensible Compound behavior with vector path and anchor and start=0
s1 = magpy.magnet.Cuboid((1,0,0), (1,1,1), [(3,0,0),(1,0,0)])
s2 = magpy.magnet.Cuboid((1,0,0), (1,1,1), [(2,0,2),(2,0,2)], R.from_rotvec([(0,0,-.1),(0,0,-.2)]))
col = magpy.Collection(s1, s2, position=[(3,0,2),(3,0,3)])
col.rotate_from_rotvec([(0,0,np.pi/2), (0,0,3*np.pi/2)], anchor=[(1,0,0), (2,0,0)], start=0)
test_po(s1, [(1,2,0),(2,1,0)], [(0,0,np.pi/2), (0,0,-np.pi/2)], (2,3))
test_po(s2, [(1,1,2),(2,0,2)], [(0,0,np.pi/2-.1), (0,0,-np.pi/2-.2)], (2,3))
test_po(col, [(1,2,2),(2,-1,3)], [(0,0,np.pi/2), (0,0,-np.pi/2)], (2,3))

# nonsensical but correct Collection behavior when col and children all have different path formats
s1 = magpy.magnet.Cuboid((1,0,0), (1,1,1), position=(1,1,1))
s2 = magpy.magnet.Cuboid((1,0,0), (1,1,1), orientation=R.from_rotvec([(0,0,-.1), (0,0,-.2)]))
col = magpy.Collection(s1, s2, position=[(1,2,3),(1,3,4)])
col.rotate_from_angax(90, 'z', anchor=(1,0,0))
test_po(s1, (0,0,1), (0,0,np.pi/2), (3,))
test_po(s2, [(1,-1,0)]*2, [(0,0,np.pi/2-.1),(0,0,np.pi/2-.2)], (2,3))
test_po(col, [(-1,0,3),(-2,0,4)], [(0,0,np.pi/2)]*2, (2,3))

# nonsensical but correct Collection behavior with vector anchor
s1 = magpy.magnet.Cuboid((1,0,0), (1,1,1), position=(1,0,1))
s2 = magpy.magnet.Cuboid((1,0,0), (1,1,1), orientation=R.from_rotvec([(0,0,-.1), (0,0,-.2)]))
col = magpy.Collection(s1, s2, position=[(3,0,3),(4,0,4)])
col.rotate_from_angax(90, 'z', anchor=[(1,0,0), (2,0,0)])
test_po(s1, [(1,0,1),(1,0,1),(2,-1,1)], [(0,0,0),(0,0,np.pi/2),(0,0,np.pi/2)], (3,3))
test_po(s2, [(0,0,0),(0,0,0),(1,-1,0),(2,-2,0)], [(0,0,-.1),(0,0,-.2),(0,0,np.pi/2-.2),(0,0,np.pi/2-.2)], (4,3))
test_po(col, [(3,0,3),(4,0,4),(1,3,4),(2,2,4)], [(0,0,0), (0,0,0), (0,0,np.pi/2), (0,0,np.pi/2)], (4,3))

# Compound rotation (anchor=None), scalar input, scalar pos
s1 = magpy.magnet.Cuboid((1,0,0), (1,1,1), (1,0,1))
s2 = magpy.magnet.Cuboid((1,0,0), (1,1,1), (0,-1,-1))
col = magpy.Collection(s1, s2)
col.rotate_from_angax(90, 'z')
test_po(s1, (0,1,1), (0,0,np.pi/2), (3,))
test_po(s2, (1,0,-1), (0,0,np.pi/2), (3,))
test_po(col, (0,0,0), (0,0,np.pi/2), (3,))

# Compound rotation (anchor=None), scalar input, vector pos, start=auto
s1 = magpy.magnet.Cuboid((1,0,0), (1,1,1), [(1,0,0),(2,0,0)])
s2 = magpy.magnet.Cuboid((1,0,0), (1,1,1), [(-1,0,0),(-2,0,0)])
col = magpy.Collection(s1, s2, position=((0,0,0), (1,0,0)))
col.rotate_from_angax(90, 'z')
test_po(s1, [(0,1,0), (1,1,0)], [(0,0,np.pi/2), (0,0,np.pi/2)], (2,3,))
test_po(s2, [(0,-1,0), (1,-3,0)], [(0,0,np.pi/2), (0,0,np.pi/2)], (2,3,))
test_po(col,[(0,0,0), (1,0,0)], [(0,0,np.pi/2), (0,0,np.pi/2)], (2,3,))

# Compound rotation (anchor=None), scalar input, vector pos, start=1
s1 = magpy.magnet.Cuboid((1,0,0), (1,1,1), [(1,0,0),(2,0,0)])
s2 = magpy.magnet.Cuboid((1,0,0), (1,1,1), [(-1,0,0),(-2,0,0)])
col = magpy.Collection(s1, s2, position=((0,0,0), (1,0,0)))
col.rotate_from_angax(90, 'z', start=1)
test_po(s1, [(1,0,0), (1,1,0)], [(0,0,0), (0,0,np.pi/2)], (2,3,))
test_po(s2, [(-1,0,0), (1,-3,0)], [(0,0,0), (0,0,np.pi/2)], (2,3,))
test_po(col,[(0,0,0), (1,0,0)], [(0,0,0), (0,0,np.pi/2)], (2,3,))

# Compound rotation (anchor=None), scalar input, vector pos, start=-1
s1 = magpy.magnet.Cuboid((1,0,0), (1,1,1), [(1,0,0),(2,0,0)])
s2 = magpy.magnet.Cuboid((1,0,0), (1,1,1), [(-1,0,0),(-2,0,0)])
col = magpy.Collection(s1, s2, position=((0,0,0), (1,0,0)))
col.rotate_from_angax(90, 'z', start=-1)
test_po(s1, [(1,0,0), (1,1,0)], [(0,0,0), (0,0,np.pi/2)], (2,3,))
test_po(s2, [(-1,0,0), (1,-3,0)], [(0,0,0), (0,0,np.pi/2)], (2,3,))
test_po(col,[(0,0,0), (1,0,0)], [(0,0,0), (0,0,np.pi/2)], (2,3,))

# Compound rotation (anchor=None), scalar input, vector pos, start->pad before
s1 = magpy.magnet.Cuboid((1,0,0), (1,1,1), [(1,0,0),(2,0,0)])
s2 = magpy.magnet.Cuboid((1,0,0), (1,1,1), [(-1,0,0),(-2,0,0)])
col = magpy.Collection(s1, s2, position=((2,0,0), (1,0,0)))
col.rotate_from_angax(90, 'z', start=-4)
test_po(s1, [(2,-1,0),(2,-1,0),(2,-1,0),(1,1,0)], [(0,0,np.pi/2)]*4, (4,3,))
test_po(s2, [(2,-3,0),(2,-3,0),(2,-3,0),(1,-3,0)], [(0,0,np.pi/2)]*4, (4,3,))
test_po(col, [(2,0,0),(2,0,0),(2,0,0),(1,0,0)], [(0,0,np.pi/2)]*4, (4,3,))

# Compound rotation (anchor=None), scalar input, vector pos, start->pad behind
s1 = magpy.magnet.Cuboid((1,0,0), (1,1,1), [(1,0,0),(2,0,0)])
s2 = magpy.magnet.Cuboid((1,0,0), (1,1,1), [(-1,0,0),(-2,0,0)])
col = magpy.Collection(s1, s2, position=((2,0,0), (1,0,0)))
col.rotate_from_angax(90, 'z', start=3)
test_po(s1, [(1,0,0),(2,0,0),(2,0,0),(1,1,0)], [(0,0,0)]*3 + [(0,0,np.pi/2)], (4,3,))
test_po(s2, [(-1,0,0),(-2,0,0),(-2,0,0),(1,-3,0)], [(0,0,0)]*3 + [(0,0,np.pi/2)], (4,3,))
test_po(col,[(2,0,0),(1,0,0),(1,0,0),(1,0,0)], [(0,0,0)]*3 + [(0,0,np.pi/2)], (4,3,))

# Compound rotation (anchor=None), vector input, simple pos, start=auto
s1 = magpy.magnet.Cuboid((1,0,0), (1,1,1), (1,0,1))
s2 = magpy.magnet.Cuboid((1,0,0), (1,1,1), (0,-1,-1))
col = magpy.Collection(s1, s2)
col.rotate_from_angax([90,-90], 'z')
test_po(s1, [(1,0,1), (0,1,1), (0,-1,1)],     [(0,0,0), (0,0,np.pi/2), (0,0,-np.pi/2)], (3,3,))
test_po(s2, [(0,-1,-1), (1,0,-1), (-1,0,-1)], [(0,0,0), (0,0,np.pi/2), (0,0,-np.pi/2)], (3,3,))
test_po(col, [(0,0,0),(0,0,0),(0,0,0)], [(0,0,0),(0,0,np.pi/2),(0,0,-np.pi/2)], (3,3))

# Compound rotation (anchor=None), vector input, vector pos, start=1
col.rotate_from_angax([-90,180], 'z', start=1)
test_po(s1, [(1,0,1), (1,0,1), (0,1,1)],      [(0,0,0), (0,0,0), (0,0,np.pi/2)], (3,3,))
test_po(s2, [(0,-1,-1), (0,-1,-1), (1,0,-1)], [(0,0,0), (0,0,0), (0,0,np.pi/2)], (3,3,))
test_po(col, [(0,0,0),(0,0,0),(0,0,0)],       [(0,0,0),(0,0,0),(0,0,np.pi/2)], (3,3))

# Compound rotation (anchor=None), vector input, vector pos, start=1, pad_behind
col.rotate_from_angax([90,180,-90], 'z', start=1)
test_po(s1, [(1,0,1), (0,1,1), (0,-1,1), (1,0,1)],       [(0,0,0), (0,0,np.pi/2), (0,0,-np.pi/2), (0,0,0)], (4,3,))
test_po(s2, [(0,-1,-1), (1,0,-1), (-1,0,-1), (0,-1,-1)], [(0,0,0), (0,0,np.pi/2), (0,0,-np.pi/2), (0,0,0)], (4,3,))
test_po(col, [(0,0,0),(0,0,0),(0,0,0),(0,0,0)],          [(0,0,0), (0,0,np.pi/2), (0,0,-np.pi/2), (0,0,0)], (4,3,))

# Compound rotation (anchor=None), vector input, simple pos, start=-3, pad_before
s1 = magpy.magnet.Cuboid((1,0,0), (1,1,1), (1,0,1))
s2 = magpy.magnet.Cuboid((1,0,0), (1,1,1), (-1,0,-1))
col = magpy.Collection(s1, s2, position=(2,0,0))
col.rotate_from_angax([90,-90], 'z', start=-3)
test_po(s1, [(2,-1,1), (2,1,1), (1,0,1)],     [(0,0,np.pi/2), (0,0,-np.pi/2), (0,0,0)], (3,3,))
test_po(s2, [(2,-3,-1), (2,3,-1), (-1,0,-1)],     [(0,0,np.pi/2), (0,0,-np.pi/2), (0,0,0)], (3,3,))
test_po(col,[(2,0,0), (2,0,0), (2,0,0)],     [(0,0,np.pi/2), (0,0,-np.pi/2), (0,0,0)], (3,3,))

# Compound rotation (anchor=None), vector input, vector pos, start=-3, pad_before AND pad_behind
s1 = magpy.magnet.Cuboid((1,0,0), (1,1,1), orientation=R.from_rotvec([(0,0,.1), (0,0,.2)]))
s2 = magpy.magnet.Cuboid((1,0,0), (1,1,1), position=[(-1,0,0), (-2,0,0)])
col = magpy.Collection(s1, s2, position=[(1,0,0), (0,0,0)])
col.rotate_from_angax([90,-90,90,-90], 'z', start=-3)
test_po(s1,
    [(1,-1,0), (1,1,0), (0,0,0), (0,0,0)],
    [(0,0,.1+np.pi/2), (0,0,.1-np.pi/2), (0,0,.2+np.pi/2), (0,0,.2-np.pi/2)],
    (4,3,))
test_po(s2,
    [(1,-2,0), (1,2,0), (0,-2,0), (0,2,0)],
    [(0,0,np.pi/2), (0,0,-np.pi/2), (0,0,np.pi/2), (0,0,-np.pi/2)],
    (4,3,))
test_po(col,
    [(1,0,0), (1,0,0), (0,0,0), (0,0,0)],
    [(0,0,np.pi/2), (0,0,-np.pi/2), (0,0,np.pi/2), (0,0,-np.pi/2)],
    (4,3,))

# CRAZY Compound rotation (anchor=None) with messy path formats
s1 = magpy.magnet.Cuboid((1,0,0), (1,1,1), orientation=R.from_rotvec([(0,0,.1), (0,0,.2)]))
s2 = magpy.magnet.Cuboid((1,0,0), (1,1,1), position=(-1,0,0))
col = magpy.Collection(s1, s2, position=[(1,0,0), (0,0,0), (3,0,3)])
col.rotate_from_angax([90,-90], 'z', start='auto')
test_po(s1,
    [(0,0,0), (0,0,0), (3,-3,0), (3,3,0)],
    [(0,0,.1), (0,0,.2), (0,0,.2+np.pi/2), (0,0,.2-np.pi/2)],
    (4,3,))
test_po(s2,
    [(-1,0,0), (3,-4,0), (3,4,0)],
    [(0,0,0), (0,0,np.pi/2), (0,0,-np.pi/2)],
    (3,3,))
test_po(col,
    [(1,0,0), (0,0,0), (3,0,3), (3,0,3), (3,0,3)],
    [(0,0,0), (0,0,0), (0,0,0), (0,0,np.pi/2), (0,0,-np.pi/2)],
    (5,3,))

```

**Comment by Alexboiboi** [comment]
> Your images are without the line 25 (`c.rotate_from_angax(45, "x")`) I guess ?

Yes, for creating the plots it was commented out

**Comment by Alexboiboi** [comment]
> @Alexboiboi please comment on this:
> 
> There are two possibilities to interpret "ambiguous" input when
> 
> 1. BaseGeo init with len(position_path)=M and len(orientation_path)=N and M!=N
> 2. Multi-anchor rotation when len(rotation_input)=M and len(anchor_input)=M and M!=N
> 
> Both cases should behave similarly in my opinion.
> 
> I propose that we follow our new "padding logic" (like when start>path_length): "If one input is longer than the other, pad the other with its edge value"
> 
> * Advantage: simple rule, simple code
> * Disadvantage: no warnings/error msg when this nonsensical input comes
> 
> Example:
> 
> ```
> src = magpy.Sensor(position=(1,2,3), orientation=R.from_rotvec((0,0,1),(0,0,2)
> print(src.position)                         # [[1,2,3]  [1,2,3]]
> print(src.orientation.as_rotvec())  # [[0,0,1]   [0,0,2]]
> ```
> 
> An alternative would be to raise a BadUserInput Exception when N!=M (unless the smaller one is 1).
> 
> * Advantage: error msg
> * Disadvantage: more restrictive, more complex code
> 
> ```
> src = magpy.Sensor(position=(1,2,3), orientation=R.from_rotvec((0,0,1),(0,0,2)
> # Error: MagpylibBadUserInput....
> ```
> 
> what so you think ?

I see three different cases:

- Case 1: position and orientation are set over move/rotate:
  -> pad up in case one of position or orientation is shorter

- Case 2: position and orientation are set over setters:
 -> cut down position/orientation to match the one which is set, if in input path length is shorter than existing one.

- Case 3: position and orientation over `__init__` : 
 -> same behavior as Case 1

**Comment by Alexboiboi** [comment]
closed by #458
